pkg/apis/core/validation/validation.go

```diff
@@ -681,8 +681,12 @@ func validateGitRepoVolumeSource(gitRepo *core.GitRepoVolumeSource, fldPath *fie
 		allErrs = append(allErrs, field.Required(fldPath.Child("repository"), ""))
 	}
 
-	pathErrs := validateLocalDescendingPath(gitRepo.Directory, fldPath.Child("directory"))
-	allErrs = append(allErrs, pathErrs...)
+	allErrs = append(allErrs, validateLocalDescendingPath(gitRepo.Directory, fldPath.Child("directory"))...)
+
+	allErrs = append(allErrs, validateNonFlagArgument(gitRepo.Repository, fldPath.Child("repository"))...)
```

Comment:
this is a tightening validation change, which means existing invalid data could get stuck, unable to be updated or deleted. what's the opinion of @kubernetes/api-approvers here?

Reply:
Right, you can't do this unless you can guarantee existing objects won't be affected.

Reply:
Alright, I'll drop this from the PR for now, and just go with the node-level validation. I'll follow up with @lavalamp about how to handle this post 1.11

Reply:
Sounds good. I'd recommend picking the kubelet changes back as well